crates/ra_assists/src/auto_import.rs

```diff
@@ -0,0 +1,780 @@
+use hir::db::HirDatabase;
+use ra_syntax::{
+    ast, AstNode, SyntaxNode, Direction, TextRange,
+    SyntaxKind::{ PATH, PATH_SEGMENT, COLONCOLON, COMMA }
+};
+use crate::assist_ctx::{AssistCtx, Assist, AssistBuilder};
+use itertools::{ Itertools, EitherOrBoth };
+
+// TODO: refactor this before merge
+mod formatting {
```

Comment:
Yeah, formatting module in ra_ide_api_light is kinda random at the moment. Really, this should be a separate crate which is as powerful as rustfmt. 

You can use `formatting` from `ra_ide_api_light`, but I would also love a separate PR which creates a new `ra_fmt` crate and moves this module there.  I think we then would be able to drop dependency from assists on ide_api_light.

Reply:
I'll go for another PR then, introducing `ra_fmt`.

Reply:
Awesome! Feel free to add/remove stuff from fmt module, it's currently a somewhat ad-hoc collections of functions, I imagine it should grow into a better "all formatting" API in the future!